fvm/evm/stdlib/checking.go

```diff
@@ -0,0 +1,283 @@
+package stdlib
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/onflow/atree"
+	"github.com/onflow/cadence"
+	"github.com/onflow/cadence/runtime"
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/cadence/runtime/interpreter"
+	"go.opentelemetry.io/otel/attribute"
+)
+
+// checkingInterface is a runtime.Interface implementation
+// that can be used for ParseAndCheckProgram.
+// It is not suitable for execution.
+type checkingInterface struct {
```

Comment:
We have a few partial implementations of the runtime interface lying around and its always very messy only implementing a few methods of it that you currently need. And every time there is an update to the interface we have to update all of them.

it would be great if cadence provided something like a `EmptyRuntimeInterface` that would do something like:

```
//...
func (inter *EmptyRuntimeInterface) GetCode(l runtime.Location) ([]byte, error) {
        if inter.GetCodeFunction != nil {
            return inter.GetCodeFunction(l)
        }
	panic("unexpected call to GetCode")
}
//...
```

that way it could be used by setting the methods, or simply by providing default panicking implementations so it can be used like this:

```
type checkingInterface struct {
	EmptyRuntimeInterface

        SystemContractCodes map[common.AddressLocation][]byte
	Programs            map[runtime.Location]*interpreter.Program
}
```

This is more of a general comment, not for. this specific PR.

Reply:
Fully agreed! Not only do we have such implementations in flow-go, but also in other projects.

That's a great idea, I'll add that to Cadence and update all repos to use it when we update to the version of Cadence that provides it 👍